onnxruntime/contrib_ops/cpu/quantization/attention_quant.cc

```diff
@@ -91,10 +100,37 @@ Status QAttention<T>::PrePack(const Tensor& weights, int input_idx, bool& is_pac
     weights_data += head_size;
   }
 
+  bool kernel_owns_prepacked_buffer = (prepacked_weight_for_caching == nullptr);
+  if (!kernel_owns_prepacked_buffer) {
+    prepacked_weight_for_caching->buffers_.push_back(std::move(packed_weights_));
```

Comment:
nit: would `bool share_prepacked_weights` be a clearer name?

Reply:
Sure done